riscv-c-api.md

```diff
@@ -203,6 +203,100 @@ function declares more than one mode or an undefined mode.
 
 This attribute is incompatible with the `naked` attribute.
 
+### `__attribute__((target("<ATTR-STRING>")))`
+
+The `target` attribute is used to enable a set of features or extensions for a
+function.
+
+For instance, you can enable the `v` extension for a specific function even if
+the `-march` or `-mcpu` options do not include the `v` extension. Importantly,
+this won't alter the global settings. Here is an example:
+
+```c
+__attribute__((target("arch=+v")))
+int foo(int a)
+{
+  return a + 5;
+}
+```
+
+Using the `target` attribute for a function should not affect the file scope
+build attributes. For example, if a file is compiled with `-march=rv64ima` and
+a function is declared with `__attribute__((target("arch=+zbb")))`, the
+`Tag_RISCV_arch` build attribute should remain `rv64ima`, not `rv64ima_zbb`.
+
+The compiler may emit a mapping symbol at the beginning of a function with the
+target attribute if the function utilizes a different set of ISA extensions.
+
+`<ATTR-STRING>` can specify the following target attributes:
+
+- `arch=`: Adds extra extensions or overrides the `-march` value specified via
+           the command line for the function.
+- `tune=`: Specifies the pipeline model and cost model associated with a
+           specific microarchitecture or core for the function.
+- `cpu=`: Specifies the pipeline mode, cost model, and extension settings for
+          the function.
+
+The interactions among the `arch`, `tune`, and `cpu` attributes mirror those of
+the `-march`, `-mtune`, and `-mcpu` options. The `cpu` attribute can be seen as
+a combination of `arch` + `tune` but holds a lower priority than the other two.
+For instance, `cpu=sifive-u74` equates to `arch=rv64gc` and
+`tune=sifive-7-series`. However, if values for `arch=` or `tune=` are provided,
+they will override the `cpu` value. Therefore, `cpu=sifive-u74;arch=rv64g` is
+equivalent to `arch=rv64g;tune=sifive-7-series`, and
+`cpu=sifive-u74;tune=sifive-5-series` is equivalent to
+`arch=rv64gc;tune=sifive-5-series`.
+
+If the same type of attribute is specified more than once, only the latest one
```

Comment:
Is this useful behavior or should we just reject if the same type appears twice? Do we envision building a string through preprocessor macros where it might not be obvious to the programmer that the same value appears twice?